src/components/Calendar/doubleCalendar/yearSelect.js

```diff
@@ -0,0 +1,64 @@
+import React, { useRef, useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { useOutsideClick } from '../../../libs/hooks';
+import RenderIf from '../../RenderIf';
+// import Select from '../../Select';
+import StyledYearLabel from './styled/yearLabel';
+import StyledSelect from './styled/select';
+
+export default function YearSelect(props) {
+    const { currentYear, yearsRange, onYearChange } = props;
+    const selectRef = useRef();
+    const [editMode, setEditMode] = useState(false);
+    const toggleEditMode = () => setEditMode(!editMode);
+    const [listenOutsideClick, stopListeningOutsideClick] = useOutsideClick(
+        selectRef,
+        toggleEditMode,
+    );
+
+    useEffect(() => {
+        if (editMode) {
+            listenOutsideClick();
+        }
+
+        return () => {
+            if (editMode) {
+                stopListeningOutsideClick();
+            }
+        };
+    }, [editMode, listenOutsideClick, stopListeningOutsideClick]);
```

Comment:
@maxxgreene @LeandroTorresSicilia @TahimiLeonBravo  here this same logic of useOutsideClick I think we should change fast.

I think we should do an internal component with the logic of the editMode